### PR TITLE
Copy rule to environment(s)

### PR DESCRIPTION
**Pull request by bryce-fitzsimons**
### Features and Changes

Allows copying a rule to 1 or more environments via new modal. Also prevents copying to invalid (disallowed) environments.

image

image

- Closes https://github.com/growthbook/growthbook/issues/2631

### CODE REVIEW

File: packages/front-end/components/Features/Rule.tsx [hunk 1]
```diff
@@ -214,36 +218,15 @@ export const Rule = forwardRef<HTMLDivElement, RuleProps>(
                 >
                   {rule.enabled ? "Disable" : "Enable"}
                 </Button>
-                {environments
-                  .filter((e) => e.id !== environment)
-                  .map((en) => (
```
Comment (jdorn) [inline packages/front-end/components/Features/Rule.tsx:219]: New orgs by default have a single `production` environment, so this button will be broken for a lot of users (not "broken" exactly, it just will open a modal that doesn't do anything).  Is it worth adding logic to only render this button if there is at least 1 other environment?

File: packages/front-end/components/Features/CopyRuleModal.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    Record<string, boolean>
+  >({});
+
+  const submit = async () => {
+    let res: { version: number } | undefined;
+    for (const env in selectedEnvironments) {
+      if (!selectedEnvironments[env]) continue;
+      for (const rule of rules) {
+        res = await apiCall<{ version: number }>(
+          `/feature/${feature.id}/${version}/rule`,
```
[the remark this answers is not in the document]
Comment (bryce-fitzsimons) [reply packages/front-end/components/Features/CopyRuleModal.tsx]: good catch, thanks for the idea. implemented this...

**Review by jdorn — APPROVED** [review]

**Comment by AtakanColak** [comment]
First of all, thank you for this feature, I have two suggestions;

1. The existing modal makes additional request per environment, which is not efficient as making a single request for the list of environments (this requires back-end changes)
2. Aligned with the initial purpose of the issue I raised; Is there a select-unselect all button in the modal?

**Comment by bryce-fitzsimons** [comment]
@AtakanColak
> 1. The existing modal makes additional request per environment, which is not efficient as making a single request for the list of environments (this requires back-end changes)
We intentionally MVP'ed it here. We're planning a different mechanism for syncing rules across environments in the next few months, so didn't want to invest too much time in this solution.

> 2. Aligned with the initial purpose of the issue I raised; Is there a select-unselect all button in the modal?
Good call out, we can add that. How many environments are you targeting typically?

**Comment by AtakanColak** [comment]
I'm not sure if I'm allowed to reveal that, but safe to say more than 10 😄 So it would be convenient to have.

**Comment by bryce-fitzsimons** [comment]
https://github.com/growthbook/growthbook/pull/2697

**Pull request merged**